plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/DeallocateVmVspCommand.java

```diff
@@ -83,9 +92,130 @@ public String getVmUuid() {
+            this._isExpungingState = isExpungingState;
+            return this;
+        }
+
+        @Override
+        public DeallocateVmVspCommand build() {
+            return new DeallocateVmVspCommand(_networkUuid,_nicFromDdUuid, _nicMacAddress, _nicIp4Address, _isL3Network, _isSharedNetwork, _vpcUuid,
+                    _networksDomainUuid, _vmInstanceName, _vmUuid, _isExpungingState);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        DeallocateVmVspCommand that = (DeallocateVmVspCommand) o;
+
+        if (_isExpungingState != that._isExpungingState) return false;
+        if (_isL3Network != that._isL3Network) return false;
+        if (_isSharedNetwork != that._isSharedNetwork) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_networksDomainUuid != null ? !_networksDomainUuid.equals(that._networksDomainUuid) : that._networksDomainUuid != null)
+            return false;
+        if (_nicFromDdUuid != null ? !_nicFromDdUuid.equals(that._nicFromDdUuid) : that._nicFromDdUuid != null)
+            return false;
+        if (_nicIp4Address != null ? !_nicIp4Address.equals(that._nicIp4Address) : that._nicIp4Address != null)
+            return false;
+        if (_nicMacAddress != null ? !_nicMacAddress.equals(that._nicMacAddress) : that._nicMacAddress != null)
+            return false;
+        if (_vmInstanceName != null ? !_vmInstanceName.equals(that._vmInstanceName) : that._vmInstanceName != null)
+            return false;
+        if (_vmUuid != null ? !_vmUuid.equals(that._vmUuid) : that._vmUuid != null) return false;
+        if (_vpcUuid != null ? !_vpcUuid.equals(that._vpcUuid) : that._vpcUuid != null) return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = _networkUuid != null ? _networkUuid.hashCode() : 0;
+        result = 31 * result + (_nicFromDdUuid != null ? _nicFromDdUuid.hashCode() : 0);
+        result = 31 * result + (_nicMacAddress != null ? _nicMacAddress.hashCode() : 0);
+        result = 31 * result + (_nicIp4Address != null ? _nicIp4Address.hashCode() : 0);
+        result = 31 * result + (_isL3Network ? 1 : 0);
+        result = 31 * result + (_isSharedNetwork ? 1 : 0);
+        result = 31 * result + (_vpcUuid != null ? _vpcUuid.hashCode() : 0);
+        result = 31 * result + (_networksDomainUuid != null ? _networksDomainUuid.hashCode() : 0);
+        result = 31 * result + (_vmInstanceName != null ? _vmInstanceName.hashCode() : 0);
+        result = 31 * result + (_vmUuid != null ? _vmUuid.hashCode() : 0);
+        result = 31 * result + (_isExpungingState ? 1 : 0);
+        return result;
+    }
```

Comment:
Please implement ``toString()`` to provide debugging state information for debugging purposes.